tmt/steps/provision/vagrant.py

```diff
@@ -282,31 +313,68 @@ def how_libvirt(self):
 
         self.plugin_install(name)
 
-        self.add_provider(name, 'memory = 1024')
-        self.vf_backup("QEMU session")
+        self.gen_virtual()
+
+        self.add_provider(name, self.kve('memory', self.data['memory']))
+        self.vf_backup("QEMU user session")
         try:
             self.add_provider(name, 'qemu_use_session = true')
         except GeneralError as error:
             # Not really an error
             #self.debug(error)
             self.vf_restore()
 
+    def how_managed(self):
+        name = 'managed'
+        self.debug("generating", name)
+
+        host = self.data['host']
+        if host is None:
+            raise SpecificationError('Remote host is not specified.')
+        self.debug("Host", host)
+
+        self.plugin_install(f"{name}-servers")
+
+        self.data['box'] = self.dummy_image
+        self.create()
+
+        self.add_provider(name, self.kve('server', host))
+
+        # Let's use the config.ssh setup first; this is backup:
+        # => override.ssh.username
+        # => override.ssh.private_key_path = ".vagrant/machines/local_linux/virtualbox/private_key"
+
+    def how_container(self):
+        self.debug("generating", "container")
+        raise SpecificationError('NYI: cannot currently run containers.')
+
     def how_openstack(self):
         self.debug("generating", "openstack")
         raise SpecificationError('NYI: cannot currently run on openstack.')
 
+    # Aliases
+    def how_remote(self):
```

Comment:
Using `remote` seems to me a bit confusing. OpenStack, Beaker, Testing Farm... provide all remote guests for testing. What about this?

    tmt run provision --how=connect --guest a.b.c.d --user login ...

How shall I provision? Connect to an existing box. This sounds to me a bit more natural / specific.

Reply:
You're right, let's use `guest` instead.